### PR TITLE
Change timeout in PluginEnableOptions to time.Duration

### PR DESCRIPTION
**Pull request by vdemeester**
This is more consistent to other options.

/cc @thaJeztah @vieux @tiborvass @anusha-ragunathan 

:frog: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by anusha-ragunathan** [comment]
Duration is not necessary here. This was discussed in https://github.com/docker/docker/pull/28595#discussion_r88989260

File: cli/command/plugin/enable.go [hunk 1]
```diff
@@ -30,7 +31,7 @@ func newEnableCommand(dockerCli *command.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.IntVar(&opts.timeout, "timeout", 0, "HTTP client timeout (in seconds)")
+	flags.DurationVar(&opts.timeout, "timeout", time.Duration(0), "HTTP client timeout (in seconds)")
```
Comment (vdemeester) [inline cli/command/plugin/enable.go]: True

**Comment by vdemeester** [comment]
> `time.Duration` lets users specify intervals more granular than seconds (ns, us, ms), which is not required here. `int64` is fine with what we want, but to be consistent, I've updated it to an `int`. 

@anusha-ragunathan I do agree that more granular intervals than seconds is not required, but from the point of view of a user, being able to use `--timeout=15s` or `--timeout=1min` makes more sence (I think :blush:). On the other hand, `docker stop timeout` is in seconds — but the `timeout` in the API struct is a `time.Duration`.

I wish we would have more consistency on the API side. I'm fine to keep an `int` on the cli side though. Any way, I put this back in design review.

/cc @tiborvass @vieux

**Comment by tiborvass** [comment]
I do agree on consistency. I would love to have time.Duration everywhere including for "stop"

**Comment by anusha-ragunathan** [comment]
@vdemeester : The downside to having more granular time periods is that the user can specify sub-second intervals for a setting that works well only with values greater than a second. So at the least, there should be some sanity check on having too small a value.

**Comment by vdemeester** [comment]
@anusha-ragunathan yeah. I think I'm gonna remove this from the milestone for now…

**Comment by LK4D4** [comment]
Sounds ok, check for >1s sounds good, too.

**Comment by anusha-ragunathan** [comment]
Looking at http client timeouts in docker, 
- trust uses 5s (cli/trust/trust.go) to connect to the notary server.
- registry auth uses 15s (registry/auth.go) to connect to the registry server.

I'm not sure what a sensible minimum timeout would be to connect to the plugin http server. Maybe we can pick 5s?

**Comment by AkihiroSuda** [comment]
Since 1.13 already shipped, I wonder we can no longer change the type, because of compatibility.

**Comment by LK4D4** [comment]
ping @vdemeester

**Comment by cpuguy83** [comment]
Should we close?

**Comment by vdemeester** [comment]
Yeah let's close for now 👼

**Pull request closed without merging**